### ISSUE TITLE
Issue running the test pipeline

### ISSUE THREAD

**Issue by guillaumecharbonnier** [Oshlack/MINTIE#40]
I have just deployed mintie using mamba and get this error. Is there some java version that need to be pinned in the conda recipe?

mintie -w -p test_params.txt cases/*.fastq.gz controls/*.fastq.gz
OpenJDK 64-Bit Server VM warning: Options -Xverify:none and -noverify were deprecated in JDK 13 and will likely be removed in a future release.                                                                                                                                 
java.lang.NoClassDefFoundError: Could not initialize class org.codehaus.groovy.vmplugin.v7.Java7                                                                                                                                                                                
        at org.codehaus.groovy.vmplugin.VMPluginFactory.<clinit>(VMPluginFactory.java:43)                                                                                                                                                                                       
        at org.codehaus.groovy.reflection.GroovyClassValueFactory.<clinit>(GroovyClassValueFactory.java:35)                                                                                                                                                                     
        at org.codehaus.groovy.reflection.ClassInfo.<clinit>(ClassInfo.java:107)                                                                                                                                                                                                
        at org.codehaus.groovy.reflection.ReflectionCache.getCachedClass(ReflectionCache.java:95)                                                                                                                                                                               
        at org.codehaus.groovy.reflection.ReflectionCache.<clinit>(ReflectionCache.java:39)                                                                                                                                                                                     
        at org.codehaus.groovy.runtime.metaclass.MetaClassRegistryImpl.registerMethods(MetaClassRegistryImpl.java:209)                                                                                                                                                                                                       
        at org.codehaus.groovy.runtime.metaclass.MetaClassRegistryImpl.<init>(MetaClassRegistryImpl.java:107)                                                                                                                                                                                                                
        at org.codehaus.groovy.runtime.metaclass.MetaClassRegistryImpl.<init>(MetaClassRegistryImpl.java:85)                                                                                                                                                                    
        at groovy.lang.GroovySystem.<clinit>(GroovySystem.java:36)                                                                      
        at org.codehaus.groovy.runtime.InvokerHelper.<clinit>(InvokerHelper.java:86)                                                                                                                                                                                                                                         
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.createCallStaticSite(CallSiteArray.java:74)                                
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.createCallSite(CallSiteArray.java:161)                                                                                                                                                                            
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)                                         
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:115)                                                                                                                                                                                                                             
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:127)                                         
        at bpipe.Runner.<clinit>(Runner.groovy:58)                                                                                                                                                                                                                              
        at bpipe.Runner9.main(Runner9.java:47)                                                                                                                                                                                                                                  
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)                                                                                                                                                                
        at java.base/java.lang.reflect.Method.invoke(Method.java:580)                                                                                                                                                                                                           
        at org.codehaus.groovy.tools.GroovyStarter.rootLoader(GroovyStarter.java:110)                                                                                                                                                                                                                                        
        at org.codehaus.groovy.tools.GroovyStarter.main(GroovyStarter.java:128)                                                                                                                                                                                                                                              
Caused by: java.lang.ExceptionInInitializerError: Exception org.codehaus.groovy.GroovyBugError [in thread "main"]                                                                                                                                                               
        at org.codehaus.groovy.vmplugin.v7.Java7.<clinit>(Java7.java:45)                                                                
        at java.base/jdk.internal.misc.Unsafe.ensureClassInitialized0(Native Method)                                                                                                                                                                                                                                         
        at java.base/jdk.internal.misc.Unsafe.ensureClassInitialized(Unsafe.java:1160)                                                  
        at java.base/jdk.internal.reflect.MethodHandleAccessorFactory.ensureClassInitialized(MethodHandleAccessorFactory.java:340)                             
        at java.base/jdk.internal.reflect.MethodHandleAccessorFactory.newConstructorAccessor(MethodHandleAccessorFactory.java:103)                             
        at java.base/jdk.internal.reflect.ReflectionFactory.newConstructorAccessor(ReflectionFactory.java:173)                                                                                                                                                                  
        at java.base/java.lang.reflect.Constructor.acquireConstructorAccessor(Constructor.java:549)                                      
        at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)                                                                                                                                                                                                                               
        at java.base/java.lang.reflect.ReflectAccess.newInstance(ReflectAccess.java:132)                                                
        at java.base/jdk.internal.reflect.ReflectionFactory.newInstance(ReflectionFactory.java:259)                                                                                                                                                                             
        at java.base/java.lang.Class.newInstance(Class.java:755)                                                                                                                                                                                                                
        at org.codehaus.groovy.vmplugin.VMPluginFactory.createPlugin(VMPluginFactory.java:57)                                                                                                                                                                                                                                
        at org.codehaus.groovy.vmplugin.VMPluginFactory.<clinit>(VMPluginFactory.java:39)                                                                                                                                                                                                                                    
        ... 20 more                                                                                                                                                                                                                                                             
java.lang.reflect.InvocationTargetException                                                                                                                                                                                                                                     
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:118)                                                                                                                                                                
        at java.base/java.lang.reflect.Method.invoke(Method.java:580)                                                                                                                                                                                                           
        at org.codehaus.groovy.tools.GroovyStarter.rootLoader(GroovyStarter.java:110)                                                                                                                                                                                                                                        
        at org.codehaus.groovy.tools.GroovyStarter.main(GroovyStarter.java:128)                                                                                                                                                                                                                                              
Caused by: java.lang.NoClassDefFoundError: Could not initialize class org.codehaus.groovy.reflection.ReflectionCache                                                                                                                                                            
        at org.codehaus.groovy.runtime.dgmimpl.NumberNumberMetaMethod.<clinit>(NumberNumberMetaMethod.java:33)                           
        at java.base/jdk.internal.misc.Unsafe.ensureClassInitialized0(Native Method)                                                                                                                                                                                            
        at java.base/jdk.internal.misc.Unsafe.ensureClassInitialized(Unsafe.java:1160)                                                                                                                                                                                          
        at java.base/jdk.internal.reflect.MethodHandleAccessorFactory.ensureClassInitialized(MethodHandleAccessorFactory.java:340)                                                                                                                                              
        at java.base/jdk.internal.reflect.MethodHandleAccessorFactory.newConstructorAccessor(MethodHandleAccessorFactory.java:103)                                                                                                                                                                                           
        at java.base/jdk.internal.reflect.ReflectionFactory.newConstructorAccessor(ReflectionFactory.java:173)                                                                                                                                                                                                               
        at java.base/java.lang.reflect.Constructor.acquireConstructorAccessor(Constructor.java:549)                                                                                                                                                                             
        at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)                                                                                                                                                                                  
        at java.base/java.lang.reflect.ReflectAccess.newInstance(ReflectAccess.java:132)                                                                                                                                                                                        
        at java.base/jdk.internal.reflect.ReflectionFactory.newInstance(ReflectionFactory.java:259)                                                                                                                                                                             
        at java.base/java.lang.Class.newInstance(Class.java:755)                                                                                                                                                                                                                
        at org.codehaus.groovy.runtime.metaclass.MetaClassRegistryImpl.createMetaMethodFromClass(MetaClassRegistryImpl.java:257)                                                                                                                                                                                             
        at org.codehaus.groovy.runtime.metaclass.MetaClassRegistryImpl.<init>(MetaClassRegistryImpl.java:110)                                                                                                                                                                                                                
        at org.codehaus.groovy.runtime.metaclass.MetaClassRegistryImpl.<init>(MetaClassRegistryImpl.java:85)                                                                                                                                                                    
        at groovy.lang.GroovySystem.<clinit>(GroovySystem.java:36)   
        at org.codehaus.groovy.runtime.InvokerHelper.<clinit>(InvokerHelper.java:86)                                                                                                                                                                                            
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.createCallStaticSite(CallSiteArray.java:74)
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.createCallSite(CallSiteArray.java:161)                                                                                                                                                                                                                         
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)                                                               
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:115)                                                                                                                                                                                                                             
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:127)                                                               
        at bpipe.Runner.<clinit>(Runner.groovy:58)                                                                                                                                                                                                                                                                           
        at bpipe.Runner9.main(Runner9.java:47)                                                                                                                
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)                                                                                                                                                                                                             
        ... 3 more                                                                                                                                                                                                                                                                                                           
Caused by: java.lang.ExceptionInInitializerError: Exception java.lang.NoClassDefFoundError: Could not initialize class org.codehaus.groovy.vmplugin.v7.Java7 [in thread "main"]
        at org.codehaus.groovy.vmplugin.VMPluginFactory.<clinit>(VMPluginFactory.java:43)                                                                                                                                                                                                                                    
        at org.codehaus.groovy.reflection.GroovyClassValueFactory.<clinit>(GroovyClassValueFactory.java:35)
        at org.codehaus.groovy.reflection.ClassInfo.<clinit>(ClassInfo.java:107)
        at org.codehaus.groovy.reflection.ReflectionCache.getCachedClass(ReflectionCache.java:95)
        at org.codehaus.groovy.reflection.ReflectionCache.<clinit>(ReflectionCache.java:39)
        at org.codehaus.groovy.runtime.metaclass.MetaClassRegistryImpl.registerMethods(MetaClassRegistryImpl.java:209)
        at org.codehaus.groovy.runtime.metaclass.MetaClassRegistryImpl.<init>(MetaClassRegistryImpl.java:107)
        ... 14 more

**Comment by mcmero** [comment]
This was addressed in https://github.com/Oshlack/MINTIE/issues/26. Please uninstall your mintie conda environment, use `conda clean` to remove old caches and then try reinstalling.

If that fails, you can also force installation of bpipe 0.9.12.

```bash
mamba install -c conda-forge -c bioconda bpipe=0.9.12
```

This version of bpipe on bioconda pins the correct version of openJDK.

**Comment by mcmero** [comment]
Thanks for reporting this, I've fixed a few issues in the latest release (this will take a while to propagate to bioconda however).

In the meantime, check your `references.groovy` file in your mintie directory (probably ` ~/.conda/envs/mintie/share/mintie-0.4.2-0`), if your `trans_fasta` parameters doesn't have a value, you will have to regenerate it:

```bash
cd ~/.conda/envs/mintie/share/mintie-0.4.2-0
file=chess3.0.fa
wget --no-check-certificate https://ccb.jhu.edu/software/stringtie/dl/gffread-0.11.6.Linux_x86_64.tar.gz
tar -xvzf gffread-0.11.6.Linux_x86_64.tar.gz && rm gffread-0.11.6.Linux_x86_64.tar.gz
gffread-0.11.6.Linux_x86_64/gffread chess3.0.gtf -g hg38.fa -w $file
```

Then update your `references.groovy` file with the path of the generated file, it should look like:

```
trans_fasta=~/.conda/envs/mintie/share/mintie-0.4.2-0/ref/chess3.0.fa
```

This is fixed in a77c636. 

The other issue (fixed in 051d771) is a change in how bpipe handles empty input parameters. To get around this, you can try installing an earlier version of bpipe:

```bash
mamba install -c conda-forge -c bioconda openjdk=11 bpipe=0.9.10 -y
```

**Comment by guillaumecharbonnier** [comment]
Many thanks for your fast fixes! I was able to complete the test workflow.
